Xamarin.Forms.Platform.Android/ButtonLayoutManager.cs

```diff
@@ -0,0 +1,271 @@
+
+						// if text is transformed, add that transformation to to ensure correct calculation of icon padding
+						if (view.TransformationMethod != null)
+							buttonText = view.TransformationMethod.GetTransformationFormatted(buttonText, view);
+
+						var measuredTextWidth = view.Paint.MeasureText(buttonText, 0, buttonText.Length());
+						var textWidth = Math.Min((int)measuredTextWidth, view.Layout.Width);
+						var contentsWidth = ViewCompat.GetPaddingStart(view) + iconWidth + view.CompoundDrawablePadding + textWidth + ViewCompat.GetPaddingEnd(view);
+
+						var newLeft = (view.MeasuredWidth - contentsWidth) / 2;
+						if (_element.ContentLayout.Position == Button.ButtonContentLayout.ImagePosition.Right)
+							newLeft = -newLeft;
+						if (ViewCompat.GetLayoutDirection(view) == ViewCompat.LayoutDirectionRtl)
+							newLeft = -newLeft;
+
+						_drawableBounds.Set(newLeft, _drawableBounds.Top, newLeft + iconWidth, _drawableBounds.Bottom);
+						drawable.Bounds = _drawableBounds;
+					}
+				}
+			}
+		}
+
+		public void OnViewAttachedToWindow(AView attachedView)
+		{
+			Update();
+		}
+
+		public void OnViewDetachedFromWindow(AView detachedView)
+		{
+		}
+
+		public void Update()
+		{
+			if (!UpdateTextAndImage())
+				UpdateImage();
+			UpdatePadding();
+		}
+
+		void OnElementChanged(object sender, VisualElementChangedEventArgs e)
+		{
+			if (_element != null)
+			{
+				_element.PropertyChanged -= OnElementPropertyChanged;
+				_element = null;
+			}
+
+			if (e.NewElement is Button button)
+			{
+				_element = button;
+				_element.PropertyChanged += OnElementPropertyChanged;
+			}
+
+			Update();
+		}
+
+		void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (_disposed || _renderer == null || _element == null)
+				return;
+
+			if (e.PropertyName == Button.PaddingProperty.PropertyName)
+				UpdatePadding();
+			else if (e.PropertyName == Button.ImageProperty.PropertyName || e.PropertyName == Button.ContentLayoutProperty.PropertyName)
+				UpdateImage();
+			else if (e.PropertyName == Button.TextProperty.PropertyName || e.PropertyName == VisualElement.IsVisibleProperty.PropertyName)
+				UpdateTextAndImage();
+			else if (e.PropertyName == Button.BorderWidthProperty.PropertyName && _borderAdjustsPadding)
+				_element.InvalidateMeasureNonVirtual(InvalidationTrigger.MeasureChanged);
+		}
+
+		void UpdatePadding()
+		{
+			if (_disposed || _renderer == null || _element == null)
+				return;
+
+			AppCompatButton view = View;
+			if (view == null)
+				return;
+
+			if (!_defaultPaddingPix.HasValue)
+				_defaultPaddingPix = new Thickness(view.PaddingLeft, view.PaddingTop, view.PaddingRight, view.PaddingBottom);
+
+			var padding = _element.Padding;
+
+			var adjustment = 0.0;
+			if (_borderAdjustsPadding && _element is IBorderElement borderElement && borderElement.IsBorderWidthSet() && borderElement.BorderWidth != borderElement.BorderWidthDefaultValue)
```

Comment:
I think we should just leave this out and not let border effect padding

Reply:
that is not used at all in this pr... I left it in because I was thinking someone may want to create a platform specific for android/ios to address #4853 
I believe all other OS use the correct layout - iOS is wrong, and Android is incorrect :) Windows is perfect ;)